MODEL-GUIDE.md

```diff
@@ -118,3 +118,119 @@ The build script will enforce the following keyword conventions.
     ```
 
 
+## x-pattern extension
+### schema
+```yaml
+x-pattern: 
+  description: >-
+    This extension is used by the bundler to generate a unique pattern schema
+    object for properties with specific a specific type, enum (if any), default.
+  type: object
+  required: [description, format]
+  properties:
+    description: 
+      description: >-
+        Description of the parent property hosting the x-pattern extension
+      type: string  
+    format:
+      description: >-
+        Controls the shape of the generated schema object.
+      type: string
+      enum:
+      - ipv4
+      - ipv6
+      - hex
+      - integer
+      - number
+      - enum
+      - string
+      - checksum
```

Comment:
do we really need `checksum` if we have `hex` ?

Reply:
Given an x-pattern with format of check - the bundler generates this checksum schema object.

// Schema Object openapi.yaml#/components/schemas/Pattern.Flow.Udp.Checksum
components:
  schemas:
    Pattern.Flow.Udp.Checksum:
      description: Checksum
      type: object
      required:
        - choice
      properties:
        choice:
          description: The type of checksum
          type: string
          enum:
            - generated
            - custom
          default: generated
        generated:
          description: A system generated checksum value
          type: string
          enum:
            - good
            - bad
          default: good
        custom:
          description: A custom checksum value
          type: integer
          format: integer_16_bits